x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -122,19 +126,40 @@ public <T extends TransportResponse> void sendRequest(
                     && securityContext.getAuthentication().getEffectiveSubject().getVersion().equals(minVersion) == false) {
                         // re-write the authentication since we want the authentication version to match the version of the connection
-                        securityContext.executeAfterRewritingAuthentication(
-                            original -> sendWithUser(
+                        securityContext.executeAfterRewritingAuthentication(original -> {
+                            try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().newStoredContext()) {
+                                AuthorizationUtils.maybePreAuthorizeChildAction(
+                                    threadPool.getThreadContext(),
+                                    connection,
+                                    request,
+                                    action,
+                                    minVersion,
+                                    clusterService.state()
+                                );
+                                sendWithUser(
+                                    connection,
+                                    action,
+                                    request,
+                                    options,
+                                    new ContextRestoreResponseHandler<>(threadPool.getThreadContext().wrapRestorable(original), handler),
+                                    sender
+                                );
+                            }
+                        }, minVersion);
+                    } else {
+                        // We need to store the thread context here to avoid that we pollute the context with parent authorization header,
+                        // which we only want to keep for a single transport request.
+                        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().newStoredContext()) {
+                            AuthorizationUtils.maybePreAuthorizeChildAction(
+                                threadPool.getThreadContext(),
                                 connection,
-                                action,
                                 request,
-                                options,
-                                new ContextRestoreResponseHandler<>(threadPool.getThreadContext().wrapRestorable(original), handler),
-                                sender
-                            ),
-                            minVersion
-                        );
-                    } else {
-                        sendWithUser(connection, action, request, options, handler, sender);
+                                action,
+                                minVersion,
+                                clusterService.state()
+                            );
+                            sendWithUser(connection, action, request, options, handler, sender);
+                        }
```

Comment:
We talked this over in a sync meeting:

My understanding is that the **persistent** pre-authz header will be set and implicitly copied to all the nodes in the same cluster (no remote cluster). There will be no additional **transient** pre-authz header.